app.js

```diff
@@ -29,5 +31,19 @@ app.use((err, req, res, next) => {
     res.status(400).send({ msg: "bad request" });
   } else next(err);
 });
+app.use((err, req, res, next) => {
+  if (err.code === "42601") {
+    res.status(400).send({ msg: "invalid body" });
+  } else next(err);
+});
+app.use((err, req, res, next) => {
+  if (err.code === "23503") {
+    res.status(404).send({ msg: "not found" });
+  } else next(err);
+});
+app.use((err, req, res, next) => {
+  console.log("error code>>", err.code);
+  next(err);
+});
```

Comment:
Might be time to consider extracting these into their own file to keep your app tidy.